framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2311,6 +2311,17 @@ public boolean shouldSuppressWarnings(@Nullable TreePath path, String errKey) {
                     // elements, because they may not have an @AnnotatedFor.
                     return false;
                 }
+                Element packageElement = elt.getEnclosingElement();
+                if (packageElement != null) {
+                    if (shouldSuppressWarnings(packageElement, errKey)) {
+                        return true;
+                    }
+                    if (isAnnotatedForThisCheckerOrUpstreamChecker(packageElement)) {
+                        // Return false immediately. Do NOT check for AnnotatedFor in the enclosing
+                        // elements, because they may not have an @AnnotatedFor.
+                        return false;
+                    }
+                }
```

Comment:
I add this addtional check in this branch because `declPath = TreePathUtil.enclosingDeclarationPath(declPath.getParentPath())` only gets to the enclosing class.